src/main/java/org/openstreetmap/atlas/checks/validation/areas/TallBuildingCheck.java

```diff
@@ -0,0 +1,723 @@
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        markAsFlagged(object.getOsmIdentifier());
+        final Map<String, String> tags = object.getOsmTags();
+
+        // level tag logic
+        if (this.hasBuildingLevelTag(tags))
+        {
+            final Optional<CheckFlag> flag = this.buildingLevelsTagFlagLogic(object, tags);
+            if (flag.isPresent())
+            {
+                return flag;
+            }
+        }
+
+        // height tag logic
+        if (this.hasHeightTag(tags))
+        {
+            final Optional<CheckFlag> flag = this.heightTagFlagLogic(object, tags);
+            if (flag.isPresent())
+            {
+                return flag;
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Determines which case the building:levels tag falls under.
+     * 
+     * @param object
+     *            building object
+     * @param tags
+     *            building object tags
+     * @return Optional CheckFlag
+     */
+    private Optional<CheckFlag> buildingLevelsTagFlagLogic(final AtlasObject object,
+            final Map<String, String> tags)
+    {
+        final double buildingLevelsTagValue;
+        try
+        {
+            buildingLevelsTagValue = Double.parseDouble(tags.get(BuildingLevelsTag.KEY));
+        }
+
+        // Case 2: Building has an invalid building:levels tag because it cannot be parsed.
+        catch (final NumberFormatException e)
+        {
+            return Optional.of(this.createFlag(object,
+                    this.getLocalizedInstruction(1, object.getOsmIdentifier())));
+        }
+
+        // Case 1: Building:levels tag greater than 100
+        if (buildingLevelsTagValue > this.maxLevelTagValue)
+        {
+            return Optional.of(this.createFlag(object,
+                    this.getLocalizedInstruction(0, object.getOsmIdentifier())));
+        }
+
+        // Don't run statistics if building contains building=apartments tag (apartments seem to
+        // be showing up a lot as outliers)
+        if (!(tags.containsKey(BuildingTag.KEY)
+                && tags.get(BuildingTag.KEY).equalsIgnoreCase(BuildingTag.APARTMENTS.toString())))
+        {
+            // Check store to see if building intersects stored entry rectangle
+            final Optional<Tuple4<String, Double, Double, Double>> rectangleStatsWhichIntersectObjectOptional = this
+                    .getRectangleStatsWhichIntersectObject(this.storedAreasWithStatistics, object,
```

Comment:
nit: `this.storedAreasWithStatistics` shouldn't need to be a param here

Reply:
addressed